sdk/python/tests/unit/infra/test_inference_unit_tests.py

```diff
@@ -244,8 +244,8 @@ def test_feature_view_inference_on_entity_value_types():
         ),
     )
 
-    # The schema is only used as a parameter, as is therefore not updated during inference.
-    assert len(feature_view_1.schema) == 1
```

Comment:
this seems like it'd be then potentially a backwards incompatible change?

Reply:
Yes I agree that the schema attribute will change behaviour. However, the comment
`# The schema is only used as a parameter, as is therefore not updated during inference.`
suggests that is should more or less have been a private attribute and that its state can be inconsistent.

I don't see a use case where someone would want to rely on the inconsistent state.

This PR tries to make this public attribute at least consistent.

Reply:
apparently this caused incompatible issue.
we use `feature_view.schema` to get feature schema, but now 
```
    def schema(self) -> List[Field]:
        return self.entity_columns + self.features
```
has duplicates.
a temp fix for us is
```
    def schema(self) -> List[Field]:
        return list(set(self.entity_columns + self.features))
```
but im not sure it's ideal or not
@gbmarc1 @adchia

Reply:
@hao-affirm good catch, let me put in a fix right now